tdp/core/runner/action_runner.py

```diff
@@ -20,16 +20,16 @@ def __init__(self, dag, executor, service_managers):
         self._executor = executor
         self._service_managers = service_managers
 
-    def run(self, action):
+    def run(self, action, action_file):
```

Comment:
We needed to remove `playbook_dir` from the ansible executor. Therefore it cannot infer what's the action's corresponding playbook anymore.

To keep the logic simple, I modified `_run_actions` to get the playbook's full path.

Indeed, we can get the playbook full path in `run` using:
```python
action_file = self.dag.collections[
    self.dag.components[action].collection_name
].actions[action]
```

Reply:
The logic you mention is in `_run_actions` so I do not understand why `run` need a new parameter.

In `run` method, you just perform a `or` with `action_file` and `action` parameter, why not put the `or` logic inside `_run_actions` and keep only one parameter for `run` ?

Reply:
I don't want to do path manipulation to extract the action's name that I put in the `ActionLong` at the end of the function. While I still need the action's path.

Reply:
I removed the logic with the `or` as it was a way meant for future usage in case we don't want ansible. But it complicates things and is not needed for now.

Reply:
I still not understand why we have 2 parameters, `action_runner` is not aware of the Ansible runner. `action` should be the only parameter and this the job of Ansible runner implementation to translate the action string into the real Ansible playbook file.

Reply:
The ansible executor should have an instance of the collections, loop through them to find the right playbook ? I don't think the executor should hold on the collections

Reply:
Maybe we have a design problem. Need time to investigate, I will not block this PR for that.